packages/gatsby/src/schema/schema.js

```diff
@@ -925,7 +925,7 @@ const addTypeToRootQuery = ({ schemaComposer, typeComposer }) => {
       },
       resolve: findManyPaginated(typeName),
     },
-  })
+  }).makeFieldNonNull([queryName, queryNamePlural])
```

Comment:
As @pieh noted `queryName` is _not_ non-nullable since there are indeed legitimate instances where a query to `someType(id: { eq: "id-of-node-that-doesnt-exist" }) { id }` could return `null`

Take for instance a series of queries one after another in `gatsby-node` whether `someType` is fetched for an `id` that may come from somewhere else. 

`queryNamePlural` can on the other hand safely be a `[Type]!`

What are your thoughts to switching over to `[Type]!` for just the plural form in this PR?

Reply:
I am now convinced that is correct, i changed the PR.